lib/java/com/google/android/material/imageview/ShapeableImageView.java

```diff
@@ -140,6 +134,24 @@ public ShapeableImageView(Context context, @Nullable AttributeSet attrs, int def
 
     attributes.recycle();
 
+    if (hasContentPadding()) {
```

Comment:
This seems not right to update paddings before layout direction is resolved? The content padding values thus got might be wrong.

Reply:
It seems that, once padding is set in `View`, `resetResolvedPadding()` ensures that `resolvePadding()` will be called when appropriate. I also can't find any documentation that `setPadding()` or `setPaddingRelative()` should only be called when layout direction is resolved.

So, I would say that the previous workflow was unnecessary. However, I might be missing something and I'm sorry if that's the case.

I couldn't reproduce any issues while testing.

Reply:
It's not about get/setPadding() but how we get content padding, which relies on View.getLayoutDirection() to decide which content padding should be used. If this call happens before the layout direction is resolved, there might be a problem.

Reply:
`View.resolvePadding()` should adjust the padding accordingly, once the layout direction is resolved.

Again, I might be missing something. Do you have an example, even if theoretical, that I can test?

Reply:
View.resolvePadding() also makes use of View.getLayoutDirection() to decide if it's RTL, which, again, may not be resolved yet in the constructor.

Reply:
This is how I see it:
- `View.resolvePadding()` is usually not called in the constructor;
- In this case, it will be called, because we need to get the current padding values (e.g., calling `super.getPaddingStart()`, etc.);
- However, setting padding also calls `View.resetResolvedPaddingInternal()`, ensuring that `View.resolvePadding()` will be called again later, when `View.getLayoutDirection()` is properly resolved;
- In the meantime, even if `View.getLayoutDirection()` is not accurate, everything will be consistent.

So, I don't see any issues that don't already occur when setting/getting the regular padding.

Reply:
So, imagine this scenario - 

If you set paddingStart and contentPaddingRight at the same time. Before RTL is resolved, in the overridden setPaddingRelative(), contentPaddingRight will be added to paddingEnd instead of paddingStart - which is wrong, because it should be added to paddingStart instead.

Reply:
That's definitely an issue. Somehow I wasn't considering cases where one would mix absolute regular padding with relative content padding, or the other way around. 👍